src/chai/ChaiMacros.hpp

```diff
@@ -11,26 +11,59 @@
 
 #include "umpire/util/Macros.hpp"
 
-#if defined(CHAI_ENABLE_CUDA) && defined(__CUDACC__)
+#if defined(CHAI_ENABLE_CUDA)
+
+#include <cuda_runtime_api.h>
 
 #define CHAI_HOST __host__
 #define CHAI_DEVICE __device__
 #define CHAI_HOST_DEVICE __device__ __host__
 
-#elif defined(CHAI_ENABLE_HIP) && defined(__HIPCC__)
+#define gpuMemcpyKind cudaMemcpyKind
+#define gpuMemcpyHostToHost cudaMemcpyHostToHost
+#define gpuMemcpyHostToDevice cudaMemcpyHostToDevice
+#define gpuMemcpyDeviceToHost cudaMemcpyDeviceToHost
+#define gpuMemcpyDeviceToDevice cudaMemcpyDeviceToDevice
+#define gpuMemcpyDefault cudaMemcpyDefault
+
+// NOTE: Cannot have if defined(__HIPCC__) in the condition below, since __HIPCC__ comes from the included header hip_runtime below.
+#elif defined(CHAI_ENABLE_HIP)
 
 #include <hip/hip_runtime.h>
 
 #define CHAI_HOST __host__
 #define CHAI_DEVICE __device__
 #define CHAI_HOST_DEVICE __device__ __host__
 
+#define gpuMemcpyKind hipMemcpyKind
+#define gpuMemcpyHostToHost hipMemcpyHostToHost
+#define gpuMemcpyHostToDevice hipMemcpyHostToDevice
+#define gpuMemcpyDeviceToHost hipMemcpyDeviceToHost
+#define gpuMemcpyDeviceToDevice hipMemcpyDeviceToDevice
+#define gpuMemcpyDefault hipMemcpyDefault
+
 #else
 
 #define CHAI_HOST
 #define CHAI_DEVICE
 #define CHAI_HOST_DEVICE
 
+#define gpuMemcpyKind int
+#define gpuMemcpyHostToHost 0
+#define gpuMemcpyHostToDevice 1
+#define gpuMemcpyDeviceToHost 2
+#define gpuMemcpyDeviceToDevice 3
+#define gpuMemcpyDefault 4
+
+#endif
+
+#if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__)
+#define __DEVICE_COMPILE__
```

Comment:
I think this should be `CHAI_DEVICE_COMPILE`

Reply:
Changed.